packages/block-library/src/button/edit.native.js

```diff
@@ -173,23 +173,33 @@ class ButtonEdit extends Component {
 
 	getBackgroundColor() {
 		const { backgroundColor, attributes } = this.props;
-		const { style } = attributes;
-
+		const { style, gradient, customGradient } = attributes;
+		const fallbackBackgroundColor = styles.defaultButton.backgroundColor;
+		const defaultGradients = SETTINGS_DEFAULTS.gradients;
+
+		if ( customGradient || gradient ) {
+			return (
+				customGradient ||
+				defaultGradients.find(
+					( defaultGradient ) => defaultGradient.slug === gradient
+				).gradient
+			);
+		}
 		return (
 			( style && style.color && style.color.background ) ||
 			backgroundColor.color ||
-			styles.fallbackButton.backgroundColor
+			fallbackBackgroundColor
 		);
 	}
 
 	getTextColor() {
 		const { textColor, attributes } = this.props;
 		const { style } = attributes;
-
+		const fallbackTextColor = styles.defaultButton.color;
```

Comment:
Do we really need that const? I would use `styles.defaultButton.color` directly in return